### PR TITLE
WIP: [stable/traefik] Keep basic auth user/passwd data in secret

### PR DESCRIPTION
**Pull request by krancour**
@c-knowles you will be interested in this, I think. It's the follow-up to #414 

This depends on https://github.com/containous/traefik/pull/1189 being merged and included in a future Traefik release.

@c-knowles if you want to test drive, the Docker image `krancour/traefik:usersfile` is built from my fork of Traefik.

### CODE REVIEW

**Comment by cknowles** [comment]
Great! Let's hope they merge that in soon. I wonder if we could cut down on some of the conditionals and always create the secret but with empty users if non are defined? We could always mount the secret too which keeps deployment.yaml simpler.

File: stable/traefik/templates/_helpers.tpl [hunk 1]
```diff
@@ -7,3 +7,7 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- define "fullname" -}}
 {{- printf "%s-%s" .Release.Name .Chart.Name | trunc 63 | trimSuffix "-" -}}
 {{- end -}}
+
+{{- define "dashboardUsers" -}}
+{{- range $key, $value := .Values.dashboard.auth.basic -}}{{- printf "%s:%s\n" $key $value -}}{{- end -}}
```
Comment (cknowles) [inline stable/traefik/templates/_helpers.tpl:12]: Does this need to be wrapped in `{{- if .Values.dashboard.auth }}`?

Comment (krancour) [reply stable/traefik/templates/_helpers.tpl:12]: Not really. The only thing that calls this is wrapped in those conditionals already and I don't foresee this being reused in this chart. (I didn't do it this way for the sake of reuse... this "sub-template" was the only way I could think of, at the time, to build a string by ranging over an array, and then get that string into a pipeline so I can base64 encode it... but I think I've thought of a more straightforward way to do that now, so I'll be updating this on Monday.)

**Comment by lachie83** [comment]
@c-knowles What's the latest on this?

**Comment by cknowles** [comment]
It looks like @krancour's PR above has been merged so I hope they include that with 1.2.0 GA. Otherwise the change looks good to me.

**Comment by lachie83** [comment]
@c-knowles do we need to wait for that release for these changes to work?

**Comment by cknowles** [comment]
@lachie83 I don't think @krancour's changes here would break anything existing but yep we'll have to wait for that next Traefik release for this to work.

**Comment by krancour** [comment]
I have some modifications I want to make to this chart still. I don't have a real sense of urgency about it because, as noted by @c-knowles I'm waiting for my PR to Traefik to make it into a GA release.

**Comment by lachie83** [comment]
@krancour @c-knowles shall we close this for now?

**Comment by krancour** [comment]
@lachie83 depends how much it bothers you to leave it open. Personally, for me, it's better to leave it open because then it doesn't fall off my own radar. Up to you tho.

**Comment by prydonius** [comment]
@krancour just checking in, we still waiting for an upstream release here?

**Comment by prydonius** [comment]
@krancour ping

**Comment by viglesiasce** [comment]
Closing as this has now been marked stale for 17 days:

https://github.com/kubernetes/charts#stale-pull-requests

**Comment by krancour** [comment]
I'll have another look at this. I'm sure the feature I contributed to Traefik (which this chart depends on) must be included in 1.3.x. Not sure when I will get to this though.

**Pull request closed without merging**